lib/image-editor/class-image-editor-crop.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Image Editor: Image_Editor_Crop class
+ *
+ * @package gutenberg
+ * @since 7.x ?
+ */
+
+/**
+ * Crop image modifier.
+ */
+class Image_Editor_Crop extends Image_Editor_Modifier {
+	/**
+	 * Distance from the left for the crop.
+	 *
+	 * @var float
+	 */
+	private $left = 0;
+
+	/**
+	 * Distance from the top for the crop.
+	 *
+	 * @var float
+	 */
+	private $top = 0;
+
+	/**
+	 * Width of the crop.
+	 *
+	 * @var float
+	 */
+	private $width = 0;
+
+	/**
+	 * Height of the crop.
+	 *
+	 * @var float
+	 */
+	private $height = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * Will populate object properties from the provided arguments.
+	 *
+	 * @param float $left   Percentage from the left for the crop.
+	 * @param float $top    Percentage from the top for the crop.
+	 * @param float $width  Percentage width for the crop.
+	 * @param float $height Percentage height for the crop.
+	 */
+	public function __construct( $left, $top, $width, $height ) {
+		$this->left   = floatval( $left );
+		$this->top    = floatval( $top );
+		$this->width  = floatval( $width );
+		$this->height = floatval( $height );
+	}
+
+	/**
+	 * Update the image metadata with the modifier.
+	 *
+	 * @access public
+	 *
+	 * @param array $meta Metadata to update.
+	 * @return array Updated metadata.
+	 */
+	public function apply_to_meta( $meta ) {
+		$meta['crop_left']   = $this->left;
+		$meta['crop_top']    = $this->top;
+		$meta['crop_width']  = $this->width;
+		$meta['crop_height'] = $this->height;
+
+		return $meta;
+	}
+
+	/**
+	 * Apply the modifier to the image
+	 *
+	 * @access public
+	 *
+	 * @param WP_Image_Editor $image Image editor.
+	 * @return bool|WP_Error True on success, WP_Error object or false on failure.
+	 */
+	public function apply_to_image( $image ) {
+		$size = $image->get_size();
+
+		$left   = round( ( $size['width'] * $this->left ) / 100.0 );
+		$top    = round( ( $size['height'] * $this->top ) / 100.0 );
+		$width  = round( ( $size['width'] * $this->width ) / 100.0 );
+		$height = round( ( $size['height'] * $this->height ) / 100.0 );
+
+		return $image->crop( $left, $top, $width, $height );
```

Comment:
Why is all the above needed? Why make it so complex?

Reply:
The image modifying, file naming, and metadata editing are all colocated on a per-modifier basis. When the modifiers can happen in any order, it's nice to have a single place to edit a modifier rather than having all the pieces interspersed with other modifiers.

Reply:
Can we just use [wp_crop_image](https://developer.wordpress.org/reference/functions/wp_crop_image/) here?

Reply:
I don't want to save to a new file just yet, so I'm using `WP_Image_Editor` directly.